File: frontend/my-app/services/productService.ts
import axiosClient from "./axiosClient";

const productService = {
  getAll: async (page: number, size: number, sortBy: string, sortDir: string) => {
    const response = await axiosClient.get(
      `/products?page=${page}&size=${size}&sortBy=${sortBy}&sortDir=${sortDir}`
    );
console.log("Product Service:", response.data.content);
    return response.data;
  },
  getById: (id: number): Promise<any> => axiosClient.get(`/products/${id}`),
  getByCategory: (categoryId: number): Promise<any> => 
    axiosClient.get(`/products/category/${categoryId}`),
  create: (product: any): Promise<any> => 
    axiosClient.post("/products", product),
  update: (id: number, product: any): Promise<any> => 
    axiosClient.put(`/products/${id}`, product),
  delete: (id: number): Promise<any> => 
    axiosClient.delete(`/products/${id}`)
};

export default productService;

File: backend/my-app/src/main/java/com/example/my_app/product/service/ProductService.java
package com.example.my_app.product.service;


import com.example.my_app.product.dto.ProductDTO;
import com.example.my_app.response.Response;
import org.springframework.data.domain.Page;

import java.util.List;

public interface ProductService {
    Response<Page<ProductDTO>> findAllProducts(int page, int size, String sortBy, String sortDir);
}
